script.js

```diff
@@ -1,4 +1,349 @@
+  }
+
+  //sorting the array, normal string sort can be applied because values are only single digit
+  //if playing with a dice of more than 2 digits, need to update this code with a compare function
+  //this will sort smallest to biggest number
+  diceRollArray.sort();
+  //reverses the array if it's playing the highest mode, hence biggest to smallest
+  if (diceMode == HIGHEST_MODE) diceRollArray.reverse();
+
+  var playerScore = "";
+
+  //string variable does not add the numbers, it concactenates the string
+  for (var i = 0; i < diceRollArray.length; i += 1) {
+    playerScore += String(diceRollArray[i]);
+  }
+
+  //assigns the current player's score to the resultArray, storing them as numbers
+  currentPlayerIndex = playerArray.indexOf(currentPlayer);
+  resultArray[currentPlayerIndex] = Number(playerScore);
+
+  rollDisplay += `<br> The ${diceMode} possible score is ${playerScore}`;
+
+  return rollDisplay;
+};
+
+/***
+ * Plays standard game and it goes on every single time the function is triggered
+ * @returns {string} Result of the current roll for output
+ */
+var playStandardGame = function () {
+  var outputMsg = `Player ${currentPlayer} has thrown the dice! <br><br>`;
+  var currentPlayerRoll = rollDice();
+
+  outputMsg += scoreRolls(currentPlayerRoll);
+
+  //initialize next Player's index
+  var nextPlayerIndex = playerArray.indexOf(currentPlayer) + 1;
+
+  //if next player's index is still within the array range, assign the next player as the current player
+  if (nextPlayerIndex < playerArray.length) {
+    currentPlayer = playerArray[nextPlayerIndex];
+    outputMsg += `<br><br> It's Player ${currentPlayer}'s turn now!`;
+    return outputMsg;
+  }
+
+  //if it's not within the array range, means all player has taken a turn, output will reflect who win
+  var winningPlayerScore = Math.max(...resultArray);
+  if (diceMode == LOWEST_MODE) winningPlayerScore = Math.min(...resultArray);
+
+  var winningPlayerIndex = resultArray.indexOf(winningPlayerScore);
+
+  outputMsg += `<br><br>With a score of ${resultArray[winningPlayerIndex]}, Player ${playerArray[winningPlayerIndex]} Wins this round!<br><br>`;
+
+  //win state, add the highscore of the winning player
+  highScoreArray[winningPlayerIndex] += 1;
+
+  outputMsg += leaderboardDisplay();
+
+  //reset resultArray
+  resetResultArray();
+
+  //also resets current player to be index 0
+  currentPlayer = playerArray[0];
+
+  outputMsg += `<br><br> Press roll dice to start a new round! It's Player ${currentPlayer}'s turn now!`;
+
+  return outputMsg;
+};
+
+/***
+ * Outputs the current leaderboard string.
+ * @returns {string} the leaderboard win string for display
+ */
+var leaderboardDisplay = function () {
+  //initialise leaderboard string
+  var leaderboardString = "Current Leaderboard:<br><br>";
+
+  //copying highscore array to manipulate stuff slice is used to make new objects in the array
+  var leaderboardArray = highScoreArray.slice();
```

Comment:
you mentioned this in your questionnaire answer already, but yes arrays cannot be copied by simply =

reason for this:
arrays are passed by reference in javascript so when you do arrA = arrB, the memory address of arrB is stored in the arrA variable (thus pass by reference). so any changes to arrA will affect arrB. 

this is in comparison to primitives like numbers, strings which are pass by value.

ways to copy an array which im sure you've come across from googling:
1. arrA = arrB.slice()
2. arrA = [...arrB]
3. arrA = JSON.parse(JSON.stringify(arrB))  - this method creates a deep copy vs a shallow copy in methods 1 and 2, which is useful if you have nested arrays/objects but has its own issues as well

more ways here: https://www.freecodecamp.org/news/how-to-clone-an-array-in-javascript-1d3183468f6a/